contracts/contract/deposit/RocketDepositPool.sol

```diff
@@ -76,7 +76,20 @@ contract RocketDepositPool is RocketBase, RocketDepositPoolInterface, RocketVaul
         require(rocketDAOProtocolSettingsDeposit.getDepositEnabled(), "Deposits into Rocket Pool are currently disabled");
         require(msg.value >= rocketDAOProtocolSettingsDeposit.getMinimumDeposit(), "The deposited amount is less than the minimum deposit size");
         RocketVaultInterface rocketVault = RocketVaultInterface(getContractAddress("rocketVault"));
-        require(rocketVault.balanceOf("rocketDepositPool").add(msg.value) <= rocketDAOProtocolSettingsDeposit.getMaximumDepositPoolSize(), "The deposit pool size after depositing exceeds the maximum size");
+        uint256 capacityNeeded = rocketVault.balanceOf("rocketDepositPool").add(msg.value);
+        if (capacityNeeded > rocketDAOProtocolSettingsDeposit.getMaximumDepositPoolSize()) {
+            // Doing a conditional require() instead of a single one optimizes for the common
+            // case where capacityNeeded fits in the deposit pool without looking at the queue
+            if (rocketDAOProtocolSettingsDeposit.getAssignDepositsEnabled()) {
+                RocketMinipoolQueueInterface rocketMinipoolQueue = RocketMinipoolQueueInterface(getContractAddress("rocketMinipoolQueue"));
+                require(capacity <= rocketDAOProtocolSettingsDeposit.getMaximumDepositPoolSize() + rocketMinipoolQueue.getEffectiveCapacity(),
+                    "The deposit pool size after depositing (and matching with minipools) exceeds the maximum size");
+            } else {
+                require(capacity <= rocketDAOProtocolSettingsDeposit.getMaximumDepositPoolSize(),
```

Comment:
Can replace this `require` with a `revert` because this condition is already tested above.